apps/dapp/src/components/Pages/Core/DappPages/Borrow/index.tsx

```diff
@@ -281,29 +304,45 @@ export const BorrowPage = () => {
   };
 
   const supply = async () => {
-    if (!signer || !wallet) return;
-    const tlcContract = new TempleLineOfCredit__factory(signer).attach(
-      env.contracts.tlc
-    );
+    if (!signer || !wallet) {
+      console.debug('Missing wallet or signer when trying to use Tlc.');
+      return;
+    }
+
     const amount = getBigNumberFromString(
       state.supplyValue,
       getTokenInfo(state.inputToken).decimals
     );
+
+    // TODO: Possibly move to signer API
```

Comment:
is this something slated for a later WIP after this pr?

Reply:
After this PR, I have an [issue](https://github.com/TempleDAO/temple/issues/1173) to examine the next phase of refactoring, with _smaller_ more targeted PRs to further iterate on this current state.